hoomd/md/TablePotentialGPU.cu

```diff
@@ -254,97 +260,105 @@ cudaError_t gpu_compute_table_forces(Scalar4* d_force,
+                    return error;
+                }
+
+            // bind the tables texture
+            tables_tex.normalized = false;
+            tables_tex.filterMode = cudaFilterModePoint;
+            error = cudaBindTexture(0, tables_tex, d_tables, sizeof(Scalar2) * table_width * table_index.getNumElements());
+            if (error != cudaSuccess)
+                return error;
             }
 
-        unsigned int run_block_size = min(block_size, max_block_size);
-
-        // setup the grid to run the kernel
-        dim3 grid( N / run_block_size + 1, 1, 1);
-        dim3 threads(run_block_size, 1, 1);
-
-        gpu_compute_table_forces_kernel<1><<< grid, threads, sizeof(Scalar4)*table_index.getNumElements() >>>(d_force,
-                                                                                                           d_virial,
-                                                                                                           virial_pitch,
-                                                                                                           N,
-                                                                                                           d_pos,
-                                                                                                           box,
-                                                                                                           d_n_neigh,
-                                                                                                           d_nlist,
-                                                                                                           d_head_list,
-                                                                                                           d_tables,
-                                                                                                           d_params,
-                                                                                                           ntypes,
-                                                                                                           table_width);
-        }
-    else
-        {
-        static unsigned int max_block_size = UINT_MAX;
-        if (max_block_size == UINT_MAX)
+        if (compute_capability < 350 && size_nlist > max_tex1d_width)
+            { // use global memory when the neighbor list must be texture bound, but exceeds the max size of a texture
+            static unsigned int max_block_size = UINT_MAX;
+            if (max_block_size == UINT_MAX)
+                {
+                cudaFuncAttributes attr;
+                cudaFuncGetAttributes(&attr, gpu_compute_table_forces_kernel<1>);
+                max_block_size = attr.maxThreadsPerBlock;
+                }
+
+            unsigned int run_block_size = min(block_size, max_block_size);
+
+            // setup the grid to run the kernel
+            dim3 grid( N / run_block_size + 1, 1, 1);
```

Comment:
instead of `N` it should be `range.second - range.first`

Reply:
should we fix this before merging?

Reply:
done